internal/api_server/server.go

```diff
@@ -99,8 +106,27 @@ func (s *Server) Run(ctx context.Context) error {
 
 	router := chi.NewRouter()
 
+	requestSizeLimiter := func() func(http.Handler) http.Handler {
+		return func(next http.Handler) http.Handler {
+			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if len(r.URL.String()) > maxUrlLength {
+					http.Error(w, fmt.Sprintf("URL too long, exceeds %d characters", maxUrlLength), http.StatusRequestURITooLong)
+					return
+				}
+				if len(r.Header) > maxNumHeaders {
+					http.Error(w, fmt.Sprintf("Request has too many headers, exceeds %d", maxNumHeaders), http.StatusRequestEntityTooLarge)
+					return
+				}
+
+				next.ServeHTTP(w, r)
+			})
+		}
+	}
+
 	// general middleware stack for all route groups
 	router.Use(
+		middleware.RequestSize(1024*1024), // Limit request body size to 1MB
+		requestSizeLimiter(),
```

Comment:
I'm wondering whether these should be before or after RequestID and Logger? Feels to me these requests should be logged, too.

Reply:
Sure